### PR TITLE
equation substeps and changegroups

### PR DESCRIPTION
**Pull request by evykassirer**
screen shot 2016-12-01 at 1 08 32 pm
screen shot 2016-12-01 at 1 09 18 pm

### CODE REVIEW

File: test/solveEquation.test.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ const MathChangeTypes = require('../lib/MathChangeTypes');
 const solveEquation = require('../lib/solveEquation');
 
 function testSolve(equationString, comparator, outputStr, debug=false) {
+  debug = true;
```
Comment (evykassirer) [inline test/solveEquation.test.js]: TODO remove

File: lib/EquationOperations.js [hunk 2]
```diff
@@ -206,16 +213,19 @@ function performTermOperationOnEquation(equation, op, term, changeType) {
...
+  return new EquationStatus(changeType, oldEquation, newEquation);
 }
 
 // Performs an operation of a term on an entire given expression
 function performTermOperationOnExpression(expression, op, term) {
   let node = (NodeType.isOperator(expression) ?
     NodeCreator.parenthesis(expression) : expression);
 
-  return NodeCreator.operator(op, [node, term]);
+  term.changeGroup = 1;
```
Comment (aelnaiem) [inline lib/EquationOperations.js]: Nice!

File: lib/solveEquation.js [hunk 3]
```diff
@@ -112,45 +112,91 @@ function step(equation, symbolName) {
...
+    }
   }
+  let substeps = [];
+  equationStatus.substeps.forEach(substep => {
+    substeps = addStep(substeps, substep, debug);
+  });
+  steps.push({
+    'changeType': equationStatus.changeType,
+    'asciimath': equationStatus.newEquation.print(),
+    'newEquation': equationStatus.newEquation,
```
Comment (aelnaiem) [inline lib/solveEquation.js]: Shouldn't we also have the oldEquation here?

Comment (evykassirer) [inline lib/solveEquation.js]: yes! I think we do want the oldNode/oldEquation for the custom changetypes that say like "add __ and __" where we'd only know the blanks from the before node/equation

Comment (aelnaiem) [inline lib/solveEquation.js]: cool!

**Pull request merged**